### PR TITLE
[Lib] Add library to read configs

### PR DESCRIPTION
**Pull request by vamahaja**
Add new format for `cephci.yaml` config which will have below major sections and library to validate and read the configs -

1. Credentials
2. Repositories
3. Packages

### CODE REVIEW

File: cephci/utils/configs.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    if not CONFIG:
+        raise ConfigError("Configuration is not passed")
+
+    try:
+        return CONFIG["credentials"]
+    except KeyError:
+        raise ConfigError("Credentials configurations are missing from config")
+
+
+def get_subscription_credentials(server):
```
Comment (pranavprakash20) [inline cephci/utils/configs.py:31]: one suggestion, the server variable here will have the value `cdn`, `stage` etc right? In that case, can we replace the server with subscription_type or something similar? Please ignore if not valid

File: cephci/utils/configs.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        _dict["password"] = _server["password"]
+        _dict["timeout"] = int(_server.get("timeout", 60))
+        _dict["retry"] = int(_server.get("retry", 6))
+    except KeyError:
+        raise ConfigError(f"Insufficient config for '{server}' in subscription")
+
+    return _dict
+
+
+def get_registry_credentials(server, build):
```
Comment (pranavprakash20) [inline cephci/utils/configs.py:48]: same here, the server variable has ibm or rh as the value, so can it be renamed as well?

File: cephci/utils/configs.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
+import yaml
```
Comment (pranavprakash20) [inline cephci/utils/configs.py:1]: also, can we have doc strings for these methods? Mentioning the params, possible values etc

Comment (vamahaja) [reply cephci/utils/configs.py:1]: Done

File: cephci/utils/configs.py (hunk 1 above)
Comment (vamahaja) [reply cephci/utils/configs.py:41]: `cdn`, `stage`  are based on the `serverurl` that's why `server` variable is used here.

File: cephci/utils/configs.py (hunk 2 above)
Comment (vamahaja) [reply cephci/utils/configs.py:66]: here server will have the values like `cdn`, `stage`, `cp` etc and build is `rh` & `ibm`

**Review by pranavprakash20 — APPROVED** [review]
LGTM

**Pull request merged**